run_test.sh

```diff
@@ -746,12 +745,16 @@ File: $(readlink -f .)/test/dkms_conf_test_invalid/dkms.conf does not represent
 EOF
 
 echo 'Testing dkms.conf with defaulted BUILT_MODULE_NAME'
-run_with_expected_error 8 dkms add test/dkms_conf_test_defaulted_BUILT_MODULE_NAME << EOF
-dkms.conf: Error! No 'BUILT_MODULE_NAME' directive specified for record #0.
-Error! Bad conf file.
-File: $(readlink -f .)/test/dkms_conf_test_defaulted_BUILT_MODULE_NAME/dkms.conf does not represent a valid dkms.conf file.
+run_with_expected_output dkms add test/dkms_conf_test_defaulted_BUILT_MODULE_NAME << EOF
+Creating symlink /var/lib/dkms/dkms_conf_test/1.0/source -> /usr/src/dkms_conf_test-1.0
+EOF
+run_with_expected_output dkms remove --all -m dkms_conf_test -v 1.0 << EOF
+Deleting module dkms_conf_test-1.0 completely from the DKMS tree.
 EOF
 
+echo 'Removing /usr/src/dkms_conf_test-1.0'
+rm -r /usr/src/dkms_conf_test-1.0
+
```

Comment:
Some years ago documentation said "BUILT_MODULE_NAME[#]=[...]If your DKMS module package contains more than one module to install, this is a required directive for all of the modules.[...]" and we have some ancient [Debian bug reports](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651973) that this does not work always ... as there is defaulting code for that variable I tried to fix that.

But since documentation says "[...]this is a required directive for all of the modules.[...]" nowadays, I'll just remove the defaulting part.

Related: in addition to "(optional)" there should be a "default=..." (or "default: unset"), at least where it's useful. And the "required" ones should be moved further up...

Reply:
That was too early to properly read the current manpage. It still lists BUILT_MODULE_NAME as optional for single module builds.

Reply:
And while we are at it, is `DEST_MODULE_LOCATION` mandatory on distributions where it is ignored? Documentation is not entirely clear.

Reply:
Something I havent tried yet, how does/should dkms behave on a (otherwise valid) dkms.conf defining 0 modules?

There is at least one package in Debian where [dkms.conf](https://sources.debian.org/src/lttng-modules/2.13.9-1/debian/lttng-modules-dkms.dkms) generates the list of modules dynamically depending on the kernel config.)

Reply:
> Related: in addition to "(optional)" there should be a "default=..." (or "default: unset"), at least where it's useful. And the "required" ones should be moved further up...

That would be amazing. Thanks in advance.

> That was too early to properly read the current manpage. It still lists BUILT_MODULE_NAME as optional for single module builds.

Hmm cannot see that in the man page. That said, since there is a precedent I'm happy and grateful for the fixes. Small request though - let's emit a warning in this case. Ideally dkms.conf like these should be fixed - looking at the [Arch r8168-dkms](https://aur.archlinux.org/cgit/aur.git/commit/dkms.conf?h=r8168-dkms&id=c29ddd8ffeb0c496de8d7d8e30b9f19fd7f3aa58) it was fixed since day 1, back in 2015

> And while we are at it, is DEST_MODULE_LOCATION mandatory on distributions where it is ignored? Documentation is not entirely clear.

Wrt `DEST_MODULE_LOCATION` I think we want to keep it mandatory and move the overrides from hard-coded logic in dkms to a setting in `/etc/dkms/framework.conf`. The logic here is that dkms module must provide sane defaults, where distros (or individuals) can change those similarly to `man (5) depmod.d` 

> how does/should dkms behave on a (otherwise valid) dkms.conf defining 0 modules?

Having a generator produce a genuine zero modules config sounds reasonable IMHO. Although I would suggest that, dkms emits a warning/info to highlight a potential bug/miss-configuration.

Reply:
>Hmm cannot see that in the man page. 

https://github.com/dell/dkms/blob/master/dkms.8.in#L492

Reply:
Re-reading it again - it doesn't say it's optional, but heavily implies it. As mentioned before improving the wording would be great. Not a requirement for this PR, so let's land these.